AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -217,16 +217,18 @@ - (void)visibilityDidChange:(BOOL)isVisible
   [super visibilityDidChange:isVisible];
 
   if (_downloaderImplementsSetPriority) {
-    ASDN::MutexLocker l(_lock);
+    _lock.lock();
     if (_downloadIdentifier != nil) {
       if (isVisible) {
         [_downloader setPriority:ASImageDownloaderPriorityVisible withDownloadIdentifier:_downloadIdentifier];
       } else {
         [_downloader setPriority:ASImageDownloaderPriorityPreload withDownloadIdentifier:_downloadIdentifier];
       }
     }
+    _lock.unlock();
```

Comment:
@garrettmoon This could be the case. I went just the safe route and lock and unlock it manually to also make it a bit more clear as we call `_updateProgressImageBlockOnDownloaderIfNeeded` afterwards that should not be called with a lock